site/en/blog/prerender-pages/index.md

```diff
@@ -0,0 +1,281 @@
+
+<figure>
+{% Img src="image/W3z1f5ZkBJSgL1V1IfloTIctbIF3/GcJd2GQ76qdXKbcUb1VD.png", alt="Screenshot of the address bar 'Typeahead' feature", width="400", height="216.5" %}
+</figure>
+
+Chrome will continually update its predictors based on your typing and selections.
+
+- For a greater than 50% confidence level (shown in amber), Chrome proactively preconnects to the domain, but does not prerender the page.
+- For a greater than 80% confidence level (shown in green), Chrome will prerender the URL.
+
+### Using the Speculation Rules API to prerender pages
+
+For the third prerender option, web developers can insert JSON instructions onto their pages to inform the browser about which URLs would benefit from prerendering:
+
+```html
+<script type="speculationrules">
+{
+  "prerender": [
+    {
+      "source": "list",
+      "urls": ["next.html", "next2.html"]
+    }
+  ]
+}
+</script>
+```
+
+The Speculation Rules API is planned to be expanded beyond this simple example with the addition of prefetch, [scores](https://github.com/WICG/nav-speculation/blob/main/triggers.md#scores) (for example, the likelihood of a navigation), and syntax to implement [document rules](https://github.com/WICG/nav-speculation/blob/main/triggers.md#document-rules) (for example, matching `href` patterns). For now, only the above syntax is supported in Chrome.
+
+For the initial launch in Chrome 108, prerender is restricted to same-origin pages, opened within the same tab. Chrome 109 plans expand this to allow [prerendering same-site cross-origin pages](https://chromestatus.com/feature/4899735257743360) (for example, `https://a.example.com` could prerender a page on `https://b.example.com`, where the other site has opted in to same-site, cross-origin prerendering). Future versions may also [enable prerendering in new tabs](https://bugs.chromium.org/p/chromium/issues/detail?id=1350676).
+
+You can feature detect Speculation Rules API support with standard HTML checks:
+
+```js
+if (HTMLScriptElement.supports && HTMLScriptElement.supports('speculationrules')) {
+  console.log('Your browser supports the Speculation Rules API.');
+}
+```
+
+Speculation rules can be:
+
+* Statically inserted in the page's HTML. For example a news media site, or a blog may prerender the newest article, if that is often the next navigation for a large proportion of users.
+* Dynamically inserted into the page by JavaScript. This could be based on application logic, personalized to the user, or on certain user actions such as hovering over, or clicking down on a link—as many libraries have done in the past with `preconnect`, `prefetch`, or even `preload`.
+
+Speculation rules can be added anywhere in the main page and do not need to be in the `<head>` element. Multiple speculation rules can also be added to the same page—adding multiple URLs in the `urls` array, or having multiple `<script type="speculationrules">` are functionally equivalent.
```

Comment:
I don't think this is enough. I think we need to spell out exactly what happens when multiple rules are added. We can either do so here or link to the explainer, but I think this is one of those things that we need to be explicit about.

Reply:
> Multiple speculation rules can also be added to the same page—adding multiple URLs in the `urls` array, or having multiple `<script type="speculationrules">` are functionally equivalent.

It sounds like this is true (based on what I see in chat and in [the spec](https://wicg.github.io/nav-speculation/speculation-rules.html#speculation-rules-script)), but I don't think it's the best way to frame this. I think it's clear to say that this:

```html
<script type="speculationrules">
{
  "prerender": [
    {
      "source": "list",
      "urls": ["one.html"]
    }
  ]
}
</script>
<script type="speculationrules">
{
  "prerender": [
    {
      "source": "list",
      "urls": ["two.html"]
    }
  ]
}
</script>
```

Is equivalent to this:

```html
<script type="speculationrules">
{
  "prerender": [
    {
      "source": "list",
      "urls": ["one.html"]
    },
    {
      "source": "list",
      "urls": ["two.html"]
    }
  ]
}
</script>
```

Reply:
And also this:

```html
<script type="speculationrules">
{
  "prerender": [
    {
      "source": "list",
      "urls": ["one.html", "two.html"]
    }
  ]
}
</script>
```

So we've three different ways of expressing this.

Reply:
Updated in https://github.com/GoogleChrome/developer.chrome.com/pull/4365/commits/47e011d666e82c908ec4e4519d5eb46d01a7f90e